src/coreclr/src/vm/eventpipebuffer.cpp

```diff
@@ -23,7 +23,7 @@ EventPipeBuffer::EventPipeBuffer(unsigned int bufferSize, EventPipeThread* pWrit
     m_state = EventPipeBufferState::WRITABLE;
     m_pWriterThread = pWriterThread;
     m_eventSequenceNumber = eventSequenceNumber;
-    m_pBuffer = new BYTE[bufferSize];
+    m_pBuffer = (BYTE*)ClrVirtualAlloc(NULL, bufferSize, MEM_COMMIT, PAGE_READWRITE);
```

Comment:
Add a comment pointing back to this issue on github so that future developers looking at this code can understand why it uses ClrVirtualAlloc. You may also want to comment that doing it this way was optimized for minimal servicing risk at acceptable performance, but there may be better performance options still available.